### PR TITLE
[11.0][mcfix_base]

### PR DESCRIPTION
**Pull request by JordiBForgeFlow**
Mcfix Base
==========

This module is part of a set of modules that are intended to make sure that
the multi-company functionality is consistent.

See https://github.com/Eficent/multicompany-fixes for the full project that we intend to move to OCA.

### CODE REVIEW

**Comment by JordiBForgeFlow** [comment]
@richard-willdooit Our work in https://github.com/Eficent/multicompany-fixes is completed for 11.0. Your review is welcome, and needed to migrate to OCA!

File: mcfix_base/README.rst [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+   :target: https://www.gnu.org/licenses/lgpl
+   :alt: License: LGPL-3
+
+==========
+Mcfix Base
+==========
+
+This module is part of a set of modules that are intended to make sure that
+the multi-company functionality is consistent.
+
```
Comment (elicoidal) [inline mcfix_base/README.rst:11]: Please add some content and details

File: mcfix_base/__manifest__.py [hunk 2]
```diff
@@ -0,0 +1,20 @@
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
```
Comment (elicoidal) [inline mcfix_base/__manifest__.py]: add standard OCA short header

File: mcfix_base/__manifest__.py [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+{
+    'name': 'Multi Company Fix Base',
+    'version': '11.0.1.0.0',
+    'summary': 'Base fixes',
+    'author': 'Eficent, Odoo Community Association (OCA)',
+    'website': 'http://www.eficent.com',
```
Comment (elicoidal) [inline mcfix_base/__manifest__.py]: target repo URL

File: mcfix_base/__manifest__.py [hunk 4]
```diff
@@ -0,0 +1,20 @@
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+{
+    'name': 'Multi Company Fix Base',
+    'version': '11.0.1.0.0',
+    'summary': 'Base fixes',
```
Comment (elicoidal) [inline mcfix_base/__manifest__.py]: improve the summary

File: mcfix_base/data/res_partner_demo.xml [hunk 5]
```diff
@@ -0,0 +1,59 @@
+<?xml version="1.0" encoding="utf-8"?>
```
Comment (elicoidal) [inline mcfix_base/data/res_partner_demo.xml]: remove encoding

File: mcfix_base/data/res_partner_demo.xml [hunk 6]
```diff
@@ -0,0 +1,59 @@
...
+<odoo>
```
Comment (elicoidal) [inline mcfix_base/data/res_partner_demo.xml]: merge odoo and data tags

File: mcfix_base/data/res_partner_demo.xml (hunk 5 above)
Comment (elicoidal) [inline mcfix_base/data/res_partner_demo.xml]: demo data in `demo` directory

File: mcfix_base/models/ir_actions_report.py [hunk 7]
```diff
@@ -0,0 +1,23 @@
+from odoo import api, models
```
Comment (elicoidal) [inline mcfix_base/models/ir_actions_report.py]: add short copyright headers

File: mcfix_base/models/ir_default.py [hunk 8]
```diff
@@ -0,0 +1,38 @@
...
+                        AND (d.user_id IS NULL OR d.user_id=u.id)
+                        AND (d.company_id IS NULL OR d.company_id={id1})
+                        AND {id2}
+                    ORDER BY d.user_id, d.company_id, d.id
+                """
+        company = self._context.get('default_company_id') or "u.company_id"
+        params = [self.env.uid, model_name]
+        if condition:
+            query = query.format(id1=company, id2="d.condition=%s")
+            # pylint: disable=E8103
```
Comment (elicoidal) [inline mcfix_base/models/ir_default.py]: can you remove following 2 comments?

File: mcfix_base/models/res_company.py [hunk 9]
```diff
@@ -0,0 +1,13 @@
+from odoo import api, models
+
+
+class Company(models.Model):
+    _inherit = 'res.company'
+
+    @api.model
+    def create(self, vals):
+        """ We need this method in order avoid the system to set the company
+        by default on a new partner. Because """
```
Comment (elicoidal) [inline mcfix_base/models/res_company.py]: remove `Because` or complete the docstring

File: mcfix_base/models/res_currency.py [hunk 10]
```diff
@@ -0,0 +1,11 @@
+from odoo import api, models
+
+
+class Currency(models.Model):
+    _inherit = "res.currency"
+
+    @api.multi
+    def amount_to_text(self, amount):
```
Comment (elicoidal) [inline mcfix_base/models/res_currency.py:11]: I am not sure why you introduce this function as part of multi-company fix: maybe you could add a comment or docstring

**Review by elicoidal — CHANGES_REQUESTED** [review]
Needs some documentation in README

File: mcfix_base/README.rst (hunk 1 above)
Comment (JordiBForgeFlow) [reply mcfix_base/README.rst:11]: Done!

File: mcfix_base/data/res_partner_demo.xml (hunk 5 above)
Comment (JordiBForgeFlow) [reply mcfix_base/data/res_partner_demo.xml]: Are you sure? I thought utf-8 was meant to be replaced only in py files.

File: mcfix_base/models/res_currency.py (hunk 10 above)
Comment (JordiBForgeFlow) [reply mcfix_base/models/res_currency.py:11]: Provided details

**Comment by JordiBForgeFlow** [comment]
@elicoidal thanks for your review. I have attended all your comments.

File: mcfix_base_model_create_hook/hooks.py [hunk 11]
```diff
@@ -0,0 +1,191 @@
...
+                'defer_parent_store_computation'):
+            if self.pool._init:
+                self.pool._init_parent[self._name] = True
+            else:
+                parent_val = vals.get(self._parent_name)
+                if parent_val:
+                    # determine parent_left: it comes right after the
+                    # parent_right of its closest left sibling
+                    pleft = None
+                    # pylint: disable=E8103
```
Comment (elicoidal) [inline mcfix_base_model_create_hook/hooks.py:113]: remove

File: mcfix_base/__manifest__.py [hunk 12]
```diff
@@ -0,0 +1,22 @@
+# Copyright 2018 Creu Blanca
+# Copyright 2018 Eficent Business and IT Consulting Services, S.L.
+# License LGPL-3.0 or later (https://www.gnu.org/licenses/lgpl).
+{
+    'name': 'Multi Company Fix Base',
+    'version': '11.0.1.0.0',
+    'summary': 'Base fixes',
```
Comment (elicoidal) [inline mcfix_base/__manifest__.py:7]: Can you improve a bit the summary?

**Review by elicoidal — APPROVED** [review]
small non blocking details

File: mcfix_base/models/ir_default.py [hunk 13]
```diff
@@ -0,0 +1,40 @@
...
+        query = """ SELECT f.name, d.json_value FROM ir_default d
+                    JOIN ir_model_fields f ON d.field_id=f.id
+                    JOIN res_users u ON u.id=%s
+                    WHERE f.model=%s
+                        AND (d.user_id IS NULL OR d.user_id=u.id)
+                        AND (d.company_id IS NULL OR d.company_id={id1})
+                        AND {id2}
+                    ORDER BY d.user_id, d.company_id, d.id
+                """
+        company = self._context.get('default_company_id') or "u.company_id"
```
Comment (richard-willdooit) [inline mcfix_base/models/ir_default.py:27]: tiny, but I prefer the variable to be company_id if it truly is (as in this case) an id, and not a company record

**Review by richard-willdooit — COMMENTED** [review]
Sorry, coming in late on this - It is looking excellent - I have only run a cursory glance at this point, but I already have a client lined up which needs this, so will be getting very involved as we deploy.  When this base and first modules are ready, expect much more from me....

**Pull request closed without merging**